src/main/java/com/mongodb/jdbc/MongoSQLConnection.java

```diff
@@ -0,0 +1,74 @@
+package com.mongodb.jdbc;
+
+import com.mongodb.ConnectionString;
+
+import java.sql.*;
+
+public class MongoSQLConnection extends MongoConnection implements Connection {
```

Comment:
General Java style question:
For all of these new hierarchies, the base `abstract class` always `implements` some JDBC interface. The subclasses `MySQL*` and `MongoSQL*` always extend that base class and therefore inherit the interface. My question is: is it stylistically frowned upon to explicitly include these `implements XYZ` declarations in the subclasses, or is it nice to have it stated explicitly?

It works with or without this so it seems to come down to style and I can't really find anything about this online.

Reply:
The few posts I see online give reasons for and against keeping the `implements` section.  
Arguments against are that it's not needed, redundant, and may affect some IDE's refactor features.  Arguments for it  are in case the parent class no longer implements the interface, it clearly shows what interface is implemented, and that JDK classes such as LinkedHashMap do it this way. 
My vote is to include the `implements interface` declarations.

Reply:
That's a good point that changing the base class to not implement the interface could lead to issues. I also vote for including them.